Source/WebCore/css/parser/CSSPropertyParserConsumer+ScrollSnap.cpp

```diff
@@ -37,20 +37,24 @@ namespace CSSPropertyParserHelpers {
 
 RefPtr<CSSValue> consumeScrollSnapType(CSSParserTokenRange& range, CSS::PropertyParserState&)
 {
-    // <'scroll-snap-type'> = none | [ x | y | block | inline | both ] [ mandatory | proximity ]?
+    // <'scroll-snap-type'> = none | [ x | y | block | inline | both ] [ mandatory | proximity ]?@(default=proximity)
     // https://drafts.csswg.org/css-scroll-snap-1/#scroll-snap-type
 
-    auto firstValue = consumeIdent<CSSValueNone, CSSValueX, CSSValueY, CSSValueBlock, CSSValueInline, CSSValueBoth>(range);
+    auto firstValue = consumeIdentRaw<CSSValueNone, CSSValueX, CSSValueY, CSSValueBlock, CSSValueInline, CSSValueBoth>(range);
     if (!firstValue)
         return nullptr;
 
+    if (*firstValue == CSSValueNone)
+        return CSSPrimitiveValue::create(CSSValueNone);
+
     // We only add the second value if it is not the initial value as described in specification
     // so that serialization of this CSSValueList produces the canonical serialization.
-    auto secondValue = consumeIdent<CSSValueProximity, CSSValueMandatory>(range);
-    if (secondValue && secondValue->valueID() != CSSValueProximity)
-        return CSSValueList::createSpaceSeparated(firstValue.releaseNonNull(), secondValue.releaseNonNull());
 
-    return CSSValueList::createSpaceSeparated(firstValue.releaseNonNull());
+    auto secondValue = consumeIdentRaw<CSSValueProximity, CSSValueMandatory>(range);
+    if (!secondValue || *secondValue == CSSValueProximity)
+        return CSSPrimitiveValue::create(*firstValue);
```

Comment:
Maybe could write one of these? But maybe it’s less logical? But also, maybe we will be deleting this code soon?
```suggestion
    if (secondValue != CSSValueMandatory)
        return CSSPrimitiveValue::create(*firstValue);
```
```suggestion
    if (secondValue.value_or(CSSValueProximity) == CSSValueProximity)
        return CSSPrimitiveValue::create(*firstValue);
```